ci/pkl-pipelines/pay-dev/e2e-helpers.pkl

```diff
@@ -0,0 +1,707 @@
+            file = "stubs-candidate-ecr-registry-test/tag" 
+          }
+          new LoadVarStep { 
+            load_var = "role" 
+            file = "assume-role/assume-role.json"
+            format = "json"
+          }
+          new LoadVarStep { 
+            load_var = "retag-role" 
+            file = "assume-retag-role/assume-role.json"
+            format = "json"
+          }
+          new LoadVarStep { 
+            load_var = "write-to-deploy-role" 
+            file = "assume-write-to-deploy-role/assume-role.json"
+            format = "json"
+          }
+          new LoadVarStep { 
+            load_var = "release_image_tag" 
+            file = "parse-candidate-tag/release-tag"
+          }
+          new LoadVarStep { 
+            load_var = "release_number" 
+            file = "parse-candidate-tag/release-number"
+          }
+        }
+      }
+
+      prepareCodeBuild("stubs", "prepare-e2e-codebuild.yml", "((.:candidate_image_tag))")
+
+      new InParallelStep {
+        in_parallel = new Listing {
+          runCodeBuild("run-codebuild-card", "card.json", 3)
+          runCodeBuild("run-codebuild-products", "products.json", 1)
+        }
+      }
+
+      new InParallelStep {
+        in_parallel = new Listing {
+          retagCandidateInEcr("stubs", "((.:release_image_tag))", "release")
+          retagCandidateInEcr("stubs", "latest", "latest")
+          retagCandidateInDockerHub(
+            "governmentdigitalservice/pay-stubs",
+            "((.:candidate_image_tag))",
+            "release"
+          )
+          copyMultiarchImageToAccount("stubs")
+        }
+      }
+    }
+    on_failure = shared_resources.paySlackNotificationForFail(
+      "#govuk-pay-starling",
+      "pay-stubs failed post-merge e2e tests"
+    )
+    on_success = shared_resources.paySlackNotificationForSuccess(
+      "#govuk-pay-activity",
+      "pay-stubs passed post-merge e2e tests and was pushed as a final release"
+    )
+  }
+
+  copyImageToEcr("postgres-15-alpine", "postgres:15-alpine")
+  copyImageToEcr("localstack-localstack-3", "localstack/localstack:2.0")
+  copyImageToEcr("selenium-standalone-chrome-3-141-59", "selenium/standalone-chrome:3-141-59")
+}
+
+// THINGS BELOW HERE MAY BE USEFUL IN SHARED_RESOURCES
+
+local function buildImage(taskName: String, context: String): TaskStep = new {
+  task = taskName
+  privileged = true
+  params {
+    ["CONTEXT"] = context
+    ["DOCKER_CONFIG"] = "docker_creds"
+  }
+  config {
+    platform = "linux"
+    image_resource {
+      type = "registry-image"
+      source { ["repository"] = "concourse/oci-build-task" }
+    }
+    inputs = new Listing {
+      new TaskConfig.Input { name = context }
+      new TaskConfig.Input { name = "docker_creds" }
+    }
+    outputs = new Listing {
+      new TaskConfig.Output { name = "image" }
+    }
+    run { path = "build" }
+  }
+}
+
+local function putCandidateImage(putName: String): PutStep = new {
```

Comment:
this can be removed or updated to below

```suggestion
local function putCandidateImage(putName: String): PutStep = putStep(putName, "image/image.tar, "tags/candidate-tag", true)
```